File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-10.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        int res = 0;
        for(String t : tokens){
            if (t.equals("+") || t.equals("-") || t.equals("*")|| t.equals("/")){
                int a = stack.pop();
                int b = stack.pop();
                if (t.equals("+")){
                    res = (a + b);
                    stack.push(res);
                }else if(t.equals("-")){
                    res = (b-a);
                    stack.push(res);
                }else if(t.equals("*")){
                    res = (a * b);
                    stack.push(res);
                }else{
                    res = (b / a);
                    stack.push(res);
                }

            }else{
                int number = Integer.parseInt(t);
                stack.push(number);
            }
        }
        return stack.peek();
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-10.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l,r = 0,0
        res = 0
        while r < len(s):
            while s[r] in s[l:r]:
                l = l + 1
            res = max(res,(r-l + 1))
            r = r + 1
        return res